### ISSUE TITLE
Remove deref trait abuse

### ISSUE THREAD

**Issue by ErnWong** [ErnWong/crystalorb#2]
E.g. https://github.com/ErnWong/crystalorb/blob/61590182c49c4dee25b7834de57aecd777b99e0d/src/timestamp.rs#L261-L272

There are probably other cases.

**Comment by smokku** [comment]
Is there a reasoning for it?
(besides it is seen as bad practice)

**Comment by ErnWong** [comment]
Not really. It's not really blocking any particular use-case or causing any headaches (yet), but it might be good to get the semantics "right".
